src/loki/search.cc

```diff
@@ -658,8 +658,15 @@ struct bin_handler_t {
         correlated.edges.erase(new_end, correlated.edges.end());
       }
 
+      //store filtered edges for retry in case find path with correlated.edges failed
+      for (auto filtered_edge : filtered) {
+          // we also add a panelty of 3600 to the score
+          filtered_edge.score = filtered_edge.score + 3600.0f;
```

Comment:
@kevinkreiser 
Hi Kelvin, the issue without adding the penalty is that, when calling ```path_algorithm->GetBestPath``` in ```thor_worker_t::get_path```, the edges will be sorted by their score first, and ```thor``` will try with the lower scores edges to find path. Then we encounter a case that ```filtered_edges``` with same score are used to ```get route```, instead of using the original edges with good heading and same score. Therefore we add a penalty such that when two edges have very close score, we make sure we tried with the edges with good heading first instead of filtered_edges. Let me know if it is unclear to you. Thanks.